dex/testing/dcrdex/harness.sh

```diff
@@ -0,0 +1,130 @@
+# used here are created by the respective harnesses.
+cat > "./markets.json" <<EOF
+{
+    "markets": [
+        {
+            "base": "DCR_simnet",
+            "quote": "BTC_simnet",
+            "epochDuration": 6000,
+            "marketBuyBuffer": 1.2
+        }
+    ],
+    "assets": {
+        "DCR_simnet": {
+            "bip44symbol": "dcr",
+            "network": "simnet",
+            "lotSize": 100000000,
+            "rateStep": 100000,
+            "feeRate": 10,
+            "swapConf": 1,
+            "fundConf": 1,
+            "configPath": "${TEST_ROOT}/dcr/alpha/dcrd.conf"
+        },
+        "BTC_simnet": {
+            "bip44symbol": "btc",
+            "network": "simnet",
+            "lotSize": 100000,
+            "rateStep": 100,
+            "feeRate": 100,
+            "swapConf": 1,
+            "fundConf": 1,
+            "configPath": "${TEST_ROOT}/btc/harness-ctl/alpha.conf"
+        }
+    }
+}
+EOF
+
+# Write dcrdex.conf.
+cat > "./dcrdex.conf" <<EOF
+regfeexpub=spubVWHTkHRefqHpw3qMN2VTWqnT9rMuqDZrbdgxp8QzGBvXCYfdUuk2VKFnsPHE1CBBNSZjrwHTx2LrM9LbXbGH7gb9qLCYc3JUkLW883SeQ2H
+pgdbname=${TEST_DB}
+simnet=1
+rpclisten=127.0.0.1:17273
+debuglevel=trace
+regfeeconfirms=1
+signingkeypass=keypass
+adminsrvon=1
+adminsrvpass=adminpass
+adminsrvaddr=127.0.0.1:16542
+EOF
+
+# Write rpc.cert and rpc.key.
+cat > "./rpc.cert" <<EOF
+-----BEGIN CERTIFICATE-----
+MIICpTCCAgagAwIBAgIQZMfxMkSi24xMr4CClCODrzAKBggqhkjOPQQDBDBJMSIw
+IAYDVQQKExlkY3JkZXggYXV0b2dlbmVyYXRlZCBjZXJ0MSMwIQYDVQQDExp1YnVu
+dHUtcy0xdmNwdS0yZ2ItbG9uMS0wMTAeFw0yMDA2MDgxMjM4MjNaFw0zMDA2MDcx
+MjM4MjNaMEkxIjAgBgNVBAoTGWRjcmRleCBhdXRvZ2VuZXJhdGVkIGNlcnQxIzAh
+BgNVBAMTGnVidW50dS1zLTF2Y3B1LTJnYi1sb24xLTAxMIGbMBAGByqGSM49AgEG
+BSuBBAAjA4GGAAQApXJpVD7si8yxoITESq+xaXWtEpsCWU7X+8isRDj1cFfH53K6
+/XNvn3G+Yq0L22Q8pMozGukA7KuCQAAL0xnuo10AecWBN0Zo2BLHvpwKkmAs71C+
+5BITJksqFxvjwyMKbo3L/5x8S/JmAWrZoepBLfQ7HcoPqLAcg0XoIgJjOyFZgc+j
+gYwwgYkwDgYDVR0PAQH/BAQDAgKkMA8GA1UdEwEB/wQFMAMBAf8wZgYDVR0RBF8w
+XYIadWJ1bnR1LXMtMXZjcHUtMmdiLWxvbjEtMDGCCWxvY2FsaG9zdIcEfwAAAYcQ
+AAAAAAAAAAAAAAAAAAAAAYcEsj5QQYcEChAABYcQ/oAAAAAAAAAYPqf//vUPXDAK
+BggqhkjOPQQDBAOBjAAwgYgCQgFMEhyTXnT8phDJAnzLbYRktg7rTAbTuQRDp1PE
+jf6b2Df4DkSX7JPXvVi3NeBru+mnrOkHBUMqZd0m036aC4q/ZAJCASa+olu4Isx7
+8JE3XB6kGr+s48eIFPtmq1D0gOvRr3yMHrhJe3XDNqvppcHihG0qNb0gyaiX18Cv
+vF8Ti1x2vTkD
+-----END CERTIFICATE-----
+EOF
+cat > "./rpc.key" <<EOF
+-----BEGIN EC PRIVATE KEY-----
+MIHcAgEBBEIADTDRCsp8om9OhJa+m46FZ5IhgLAno1Rp6B0i2lqESL5x9vV/upiV
+TbNzCeFqEY5/Ujra9f8ZovqMlrIQmNOaZFmgBwYFK4EEACOhgYkDgYYABAClcmlU
+PuyLzLGghMRKr7Fpda0SmwJZTtf7yKxEOPVwV8fncrr9c2+fcb5irQvbZDykyjMa
+6QDsq4JAAAvTGe6jXQB5xYE3RmjYEse+nAqSYCzvUL7kEhMmSyoXG+PDIwpujcv/
+nHxL8mYBatmh6kEt9Dsdyg+osByDRegiAmM7IVmBzw==
+-----END EC PRIVATE KEY-----
+EOF
+
+# DEX admin script
+cat > "${DCRDEX_DATA_DIR}/dexadm" <<EOF
+#!/bin/sh
+curl --cacert ${DCRDEX_DATA_DIR}/rpc.cert --basic -u u:adminpass https://127.0.0.1:16542/api/\$@
+EOF
+chmod +x "${DCRDEX_DATA_DIR}/dexadm"
```

Comment:
Initial idea here was to export a function:
```
dexadm() {
    curl --cacert somewhere/rpc.cert --basic -u u:adminpass https://127.0.0.1:16542/api/$1
}
export -f dexadm
```

But there are 2 issues with that approach:
- function export is a bash-only feature, I remember reading a convo somewhere where bash scripting was discouraged in dcr projects
- more importantly though, any exported variable or function only takes effect in the current terminal window which is shortly replaced by the tmux window. When I tested it, any exported variable is only available before attaching the tmux window.

The only other thing I think can be done here is to add a tmux ctrl window where a tester can more easily fiddle with the `dexadm` script. Ofc, the script can be executed from any other terminal window regardless.

Reply:
I certainly do not discourage bash scripting.  I encourage it.  I think there are (t)csh fans abound, but that's their preference.  bash is by far the most common shell.

Anyway, putting it into `"${DCRDEX_DATA_DIR}/dexadm"` is a fine solution.